programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -1541,70 +1529,26 @@ declare_builtin_function!(
         }
 
         if let Some(instruction_context) = found_instruction_context {
-            let result_header = translate_type_mut::<ProcessedSiblingInstruction>(
+            translate_mut!(
                 memory_mapping,
-                meta_addr,
                 invoke_context.get_check_aligned(),
-            )?;
+                let result_header: &mut ProcessedSiblingInstruction = map(meta_addr)?;
+            );
+            let ProcessedSiblingInstruction { data_len, accounts_len } = *result_header;
 
-            if result_header.data_len == (instruction_context.get_instruction_data().len() as u64)
-                && result_header.accounts_len
+            if data_len == (instruction_context.get_instruction_data().len() as u64)
+                && accounts_len
                     == (instruction_context.get_number_of_instruction_accounts() as u64)
             {
-                let program_id = translate_type_mut::<Pubkey>(
-                    memory_mapping,
-                    program_id_addr,
-                    invoke_context.get_check_aligned(),
-                )?;
-                let data = translate_slice_mut::<u8>(
-                    memory_mapping,
-                    data_addr,
-                    result_header.data_len,
-                    invoke_context.get_check_aligned(),
-                )?;
-                let accounts = translate_slice_mut::<AccountMeta>(
+                let _ = result_header; // drop(result_header) and re-map it
```

Comment:
Why not call `drop`?

Reply:
Can't drop references explicitly. This is what the Rust compiler suggests instead.